### PR TITLE
Further document item metatable

### PR DESCRIPTION
**Pull request by SleepyMode**

### CODE REVIEW

File: gamemode/core/meta/sh_item.lua [hunk 1]
```diff
@@ -45,39 +45,60 @@ function ITEM:GetID()
...
+-- @realm shared
+-- @treturn number The skin of the item
 function ITEM:GetSkin()
 	return self.skin or 0
 end
 
+--- Deprecated function which currently returns nothing.
+-- @realm shared
+-- @treturn nil Nothing
 function ITEM:GetMaterial()
```
Comment (impulsh) [inline gamemode/core/meta/sh_item.lua]: Probably shouldn't add documentation for deprecated methods

File: gamemode/core/meta/sh_item.lua [hunk 2]
```diff
@@ -86,7 +107,8 @@ function ITEM:Print(detail)
 	end
 end
 
--- Dev Buddy, You don't have to make another function to print the item Data.
+--- A utility function printing all information about the item.
+-- @realm shared
 function ITEM:PrintData()
```
Comment (impulsh) [inline gamemode/core/meta/sh_item.lua]: Should be more clear that this method prints specifically the item's data

File: gamemode/core/meta/sh_item.lua (hunk 1 above)
Comment (SleepyMode) [reply gamemode/core/meta/sh_item.lua]: Done

File: gamemode/core/meta/sh_item.lua (hunk 2 above)
Comment (SleepyMode) [reply gamemode/core/meta/sh_item.lua]: Also done

**Comment by SleepyMode** [comment]
Seems like the documentation's now passing without any issues.

Big thanks to @impulsh for helping me pinpoint the error, being me using `table` as the type instead of `tab`.

All requested changes have been fixed as well, hope that'll cut it now?

**Pull request merged**